pkg/tests/apis/alerting/rules/compat/alertrule_test.go

```diff
@@ -0,0 +1,421 @@
+package compat
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/grafana/grafana/apps/alerting/rules/pkg/apis/alerting/v0alpha1"
+	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/tests/api/alerting"
+	"github.com/grafana/grafana/pkg/tests/apis/alerting/rules/common"
+	"github.com/grafana/grafana/pkg/util"
+	prom_model "github.com/prometheus/common/model"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestIntegrationAlertRuleCompatCreateViaK8s(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	ctx := context.Background()
+	helper := common.GetTestHelper(t)
+
+	k8sClient := common.NewAlertRuleClient(t, helper.Org1.Admin)
+
+	legacyClient := alerting.NewAlertingLegacyAPIClient(helper.GetListenerAddress(), "admin", "admin")
+
+	// Ensure the old provisioning API is enabled
+	allRules, status, _ := legacyClient.GetAllRulesWithStatus(t)
+	require.Equal(t, 200, status)
+	require.NotNil(t, allRules)
+
+	// Create test folder first
+	common.CreateTestFolder(t, helper, "test-folder")
+
+	rule := ngmodels.RuleGen.With(
+		ngmodels.RuleMuts.WithUniqueUID(),
+		ngmodels.RuleMuts.WithUniqueTitle(),
+		ngmodels.RuleMuts.WithNamespaceUID("test-folder"),
+		ngmodels.RuleMuts.WithGroupName("test-group"),
+		ngmodels.RuleMuts.WithIntervalMatching(time.Duration(10)*time.Second),
+	).Generate()
+
+	alertRule := &v0alpha1.AlertRule{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "default",
+			Annotations: map[string]string{
+				"grafana.app/folder": "test-folder",
+				// use provenance api to allow use of the provisioning api
+				"grafana.app/provenance": string(ngmodels.ProvenanceAPI),
+			},
+		},
+		Spec: v0alpha1.AlertRuleSpec{
+			Title: rule.Title,
+			Data: map[string]v0alpha1.AlertRuleQuery{
+				"A": {
+					QueryType:     "query",
+					DatasourceUID: v0alpha1.AlertRuleDatasourceUID(rule.Data[0].DatasourceUID),
+					Model:         rule.Data[0].Model,
+					Source:        util.Pointer(true),
+					RelativeTimeRange: &v0alpha1.AlertRuleRelativeTimeRange{
+						From: v0alpha1.AlertRulePromDurationWMillis("5m"),
+						To:   v0alpha1.AlertRulePromDurationWMillis("0s"),
+					},
+				},
+			},
+			Trigger: v0alpha1.AlertRuleIntervalTrigger{
+				Interval: v0alpha1.AlertRulePromDuration(fmt.Sprintf("%ds", rule.IntervalSeconds)),
+			},
+		},
+	}
+
+	created, err := k8sClient.Create(ctx, alertRule, v1.CreateOptions{})
+	require.NoError(t, err)
+	require.NotNil(t, created)
+
+	t.Run("should be able to use the provisioning API with this rule", func(t *testing.T) {
```

Comment:
Just a food for thoughts. I wonder if a better test could be defining full payload and expected responses  in files and then just create and compare JSONs. In other words, we create test-data files describe k8s resource in JSON and then expected responses from regular API and provisioning APIs. Then the test is just create a resource and compare responses of two APIs with those files.